web/controllers/project_user_controller.ex

```diff
@@ -27,4 +29,25 @@ defmodule CodeCorps.ProjectUserController do
   def handle_update(_conn, model, attributes) do
     model |> ProjectUser.update_changeset(attributes)
   end
+
+  defp schedule_email(conn, _) do
+    conn
+    |> Plug.Conn.register_before_send(&maybe_send_email/1)
+  end
+
+  defp maybe_send_email(%{assigns: %{ changeset: changeset}} = conn) do
+    case {changeset |> Ecto.Changeset.get_change(:role), changeset.data.role} do
+      {"contributor", "pending"} -> send_acceptance_email(conn.assigns.project_user)
+      _ -> nil
+    end
+
+    conn
+  end
+  defp maybe_send_email(conn), do: conn
```

Comment:
This function actually defines behavior which is not exactly basic. To paraphrase it:

If the changeset changes a role that was `pending` to a role that is `contributor`, send acceptance email, otherwise do nothing.

I think that warrants a test and would probably be more easily testable if we put it in a separate module.

Not sure if you wanna do it here or in a separate PR, but we tend to delay these things and I feel we should probably strive to do that less.